dss/index/es/document.py

```diff
@@ -213,7 +213,8 @@ def _prepare_index(self, dryrun):
             IndexManager.create_index(es_client, self.replica, index_name)
         return index_name
 
-    def get_shape_descriptor(self) -> typing.Optional[str]:
+    @staticmethod
+    def _get_shape_descriptor(document) -> typing.Optional[str]:
```

Comment:
A static/class method of a class whose sole argument is an instance of that class should definitely be an instance method. Why is conversion necessary? I surely isn't intuitive.

Reply:
My intention was to discourage a possible bug that could arise from the possibility that `doc._get_shape_descriptor() != doc['shape_descriptor']`, since `doc` is mutable.

This has been backed out.